### PR TITLE
Final group capstone - Book an Appointment

### PR DESCRIPTION
**Pull request by zuhebahmed88091**
# Dear Reviewer,

Thank you for reviewing our project.

---

## Total Team Members: 4

@zuhebahmed88091 
@marvin-nyalik 
@MesakDuduCoder 
@Zafron047

## Description

This pull request implements a feature-rich Tour Booking application based on the provided requirements. The app provides the following features:

1. Navigation Panel:
   - "Packages" link to view a list of available tour packages.
   - "New Reservations" link to add new reservations.
   - "List of Reservations" link to view a list of reservations.
   - "Delete Reservation" link to delete reservations.
   - "Add tour package" link (visible to admins) to add new tour packages.
   - "Delete tour package" link (visible to admins) to delete existing tour packages.

2. Main Page:
   - Displays a list of tour packages.
   - Allows users to click the "Book" button to reserve a package.

3. Packages Page:
   - Displays a list of all available tour packages.
   
4. Tour Package Details Page:
   - Provides detailed information about a selected tour package, including package types and additional details.
   - Allows users to click the "Book" button to reserve the package.

5. Admin Features:
   - "Add package" link allows admins to add new tour packages.
   - "Delete package" link allows admins to view and delete tour packages.

6. Reservations Page:
   - Allows users to view a list of their reservations, including package, date, and city information.

7. Reservation Details Page:
   - Provides more details about a selected reservation.

8. List of Reservations Page:
   - Allows users to view a list of all reservations.
   - Provides a "Delete" button to remove reservations.

9. Cancellation Feature:
   - Users can cancel a reservation, marking it as removed and removing it from the main list.

10. Reservation Booking:
    - Users can book reservations by clicking the "Reservation +" link.
    - They can add a city name, select a date, and choose a tour package.
    - Package type radio buttons become available for selection.

11. Responsive Design:
    - The application is responsive and works seamlessly on both mobile and desktop devices.

## Testing

The implemented features have been thoroughly tested to ensure functionality, responsiveness, and data consistency.

## Additional Notes

The code adheres to best practices and maintainability standards. All functionality and interactions meet the specified requirements.

Please review this pull request, and your feedback and suggestions are welcome.

## Checklist

- [ ] Code adheres to project coding standards.
- [ ] Unit tests have been added or updated as needed.
- [ ] Documentation has been updated, if necessary.
- [ ] The code has been reviewed and tested by team members.
- [ ] No regressions or breaking changes have been introduced.

Thank you for your consideration.

For more details about the requirements please click [HERE](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md)

Best regards,
@marvin-nyalik @zuhebahmed88091 @MesakDuduCoder @Zafron047

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,114 @@
...
+
+### Run tests
+
+To run tests, run the following command: 
+
+Example command:
+
+```sh
+  npm test
+```
```
Comment (Meltrust) [inline README.md:106]: -  [ ] 2 tests are not passing:

image


Kindly, fix those 2 tests in order to preserve your app's integrity. 👍

File: README.md [hunk 2]
```diff
@@ -1,7 +1,114 @@
-# Tour Savvy App
+# Tour Savvy App Frontend
```
Comment (Meltrust) [inline README.md:1]: - [ ] In the "add package" section:

image

This isn't very clear.  One says "Add" and the other "Create".  What button should the user push?  Kindly rename the buttons or add clarification. 👍

File: README.md [hunk 3]
```diff
@@ -1,7 +1,114 @@
...
+<a name="readme-top"></a>
```
Comment (Meltrust) [inline README.md:2]: - [ ] Kindly , refactor your functions to ES6 arrow functions so your code complies with the latest standards (I found 4):

image

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

 Shouldn't take you more than 5-10 min so don't worry.  That way, your code will meet the latest standards. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @zuhebahmed88091,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Most tests are passing✔️
-  Nice code organization ✔️ 
-  Nice frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md (hunk 1 above)
Comment (Zafron047) [reply README.md:106]: ## @Meltrust 
jest


Swiper latest version was conflicting with jest. Which we have now solve by downgrading to stable version 9.
Thanks for the feedback 👍 😄

File: README.md (hunk 2 above)
Comment (Zafron047) [reply README.md:1]: @Meltrust 
We have worked on removing the confusion while add and create seems same from the user endpoint.

addpackage

Replaced create with a submit button. 😄 

Thanks for the feedback.

File: README.md (hunk 3 above)
Comment (Zafron047) [reply README.md:2]: @Meltrust 

We have changed all the functions to es6 which is the modern practice. Thank you for improving our code.

👍 😺

**Review by Meltrust — APPROVED** [review]
#### Hi @zuhebahmed88091,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- The "Add" section is now clearer✔️ 
- Functions are now ES6✔️
- All tests are now passing ✔️  
- Superb job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**